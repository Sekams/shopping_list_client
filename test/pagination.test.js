import React from 'react';
import Pagination from '../src/components/pagination'

test("renders the pagination component", () => {
    global.localStorage.setItem("pageLimit", 6);
    global.localStorage.setItem("searchTerm", "f");
    const props = {
        items: [1, 2, 3, 4, 5, 6],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    expect(wrapper).toMatchSnapshot();
});

test("handle all clicks", () => {
    global.localStorage.setItem("pageLimit", 6);
    global.localStorage.setItem("searchTerm", "");
    const props = {
        items: [1, 2, 3, 4, 5, 6],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    expect(wrapper).toMatchSnapshot();
});

test("handles no pages", () => {
    global.localStorage.setItem("pageLimit", 6);
    global.localStorage.setItem("searchTerm", "");
    const props = {
        items: [1, 2, 3],
        total_items: 3,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    expect(wrapper).toMatchSnapshot();
});

test("handle all clicks", () => {
    global.localStorage.setItem("pageLimit", 6);
    global.localStorage.setItem("searchTerm", "");
    const props = {
        items: [1, 2, 3, 4, 5, 6],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    expect(wrapper).toMatchSnapshot();
});

test("handle pages", () => {
    global.localStorage.setItem("pageLimit", 1);
    global.localStorage.setItem("searchTerm", "");
    const props = {
        items: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    expect(wrapper).toMatchSnapshot();
});

test("handle pages", () => {
    global.localStorage.setItem("pageLimit", 1);
    global.localStorage.setItem("searchTerm", "");
    const props = {
        items: [4, 5, 6],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    const prevProps = {
        items: [1, 2, 3],
        total_items: 16,
        onChangePage: jest.fn()
    }
    wrapper.instance().componentDidUpdate(prevProps, wrapper.instance().state);
    expect(wrapper).toMatchSnapshot();
});

test("handle fetch", () => {
    global.localStorage.setItem("pageLimit", 6);
    global.localStorage.setItem("searchTerm", "");
    fetch.mockResponseOnce(JSON.stringify(
        {
            "message": "Shopping lists found",
            "shoppingLists": [
                {
                    "created_on": "Fri, 17 Nov 2017 22:45:13 GMT",
                    "id": 1,
                    "modified_on": "Fri, 17 Nov 2017 22:45:13 GMT",
                    "title": "Groceries",
                    "user_id": 1
                },
                {
                    "created_on": "Fri, 17 Nov 2017 22:45:42 GMT",
                    "id": 2,
                    "modified_on": "Fri, 17 Nov 2017 22:45:42 GMT",
                    "title": "Food",
                    "user_id": 1
                }
            ],
            "status": "success",
            "total": 2
        }),
        {
            status: 200,
            ok: true
        }
    );
    const props = {
        items: [4, 5, 6],
        total_items: 16,
        onChangePage: jest.fn()
    }
    const wrapper = shallow(
        <Pagination {...props} />
    );
    expect(wrapper).toMatchSnapshot();
});