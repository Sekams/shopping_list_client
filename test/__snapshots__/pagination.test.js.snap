// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle all clicks 1`] = `
<ul
  className="pagination pagination-list"
>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      ««
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      «
    </a>
  </li>
  <li
    className=""
    key="0"
  >
    <a
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className=""
    key="1"
  >
    <a
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="active"
    key="2"
  >
    <a
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »»
    </a>
  </li>
</ul>
`;

exports[`handle all clicks 2`] = `
<ul
  className="pagination pagination-list"
>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      ««
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      «
    </a>
  </li>
  <li
    className=""
    key="0"
  >
    <a
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className=""
    key="1"
  >
    <a
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="active"
    key="2"
  >
    <a
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »»
    </a>
  </li>
</ul>
`;

exports[`handle fetch 1`] = `
<ul
  className="pagination pagination-list"
>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      ««
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      «
    </a>
  </li>
  <li
    className="active"
    key="0"
  >
    <a
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className=""
    key="1"
  >
    <a
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className=""
    key="2"
  >
    <a
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      »
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      »»
    </a>
  </li>
</ul>
`;

exports[`handle pages 1`] = `
<ul
  className="pagination pagination-list"
>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      ««
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      «
    </a>
  </li>
  <li
    className=""
    key="0"
  >
    <a
      onClick={[Function]}
    >
      7
    </a>
  </li>
  <li
    className=""
    key="1"
  >
    <a
      onClick={[Function]}
    >
      8
    </a>
  </li>
  <li
    className=""
    key="2"
  >
    <a
      onClick={[Function]}
    >
      9
    </a>
  </li>
  <li
    className=""
    key="3"
  >
    <a
      onClick={[Function]}
    >
      10
    </a>
  </li>
  <li
    className=""
    key="4"
  >
    <a
      onClick={[Function]}
    >
      11
    </a>
  </li>
  <li
    className=""
    key="5"
  >
    <a
      onClick={[Function]}
    >
      12
    </a>
  </li>
  <li
    className=""
    key="6"
  >
    <a
      onClick={[Function]}
    >
      13
    </a>
  </li>
  <li
    className=""
    key="7"
  >
    <a
      onClick={[Function]}
    >
      14
    </a>
  </li>
  <li
    className=""
    key="8"
  >
    <a
      onClick={[Function]}
    >
      15
    </a>
  </li>
  <li
    className="active"
    key="9"
  >
    <a
      onClick={[Function]}
    >
      16
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      »»
    </a>
  </li>
</ul>
`;

exports[`handle pages 2`] = `
<ul
  className="pagination pagination-list"
>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      ««
    </a>
  </li>
  <li
    className="disabled"
  >
    <a
      onClick={[Function]}
    >
      «
    </a>
  </li>
  <li
    className="active"
    key="0"
  >
    <a
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className=""
    key="1"
  >
    <a
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className=""
    key="2"
  >
    <a
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className=""
    key="3"
  >
    <a
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className=""
    key="4"
  >
    <a
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className=""
    key="5"
  >
    <a
      onClick={[Function]}
    >
      6
    </a>
  </li>
  <li
    className=""
    key="6"
  >
    <a
      onClick={[Function]}
    >
      7
    </a>
  </li>
  <li
    className=""
    key="7"
  >
    <a
      onClick={[Function]}
    >
      8
    </a>
  </li>
  <li
    className=""
    key="8"
  >
    <a
      onClick={[Function]}
    >
      9
    </a>
  </li>
  <li
    className=""
    key="9"
  >
    <a
      onClick={[Function]}
    >
      10
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      »
    </a>
  </li>
  <li
    className=""
  >
    <a
      onClick={[Function]}
    >
      »»
    </a>
  </li>
</ul>
`;

exports[`handles no pages 1`] = `""`;

exports[`renders the pagination component 1`] = `""`;
